src/storage-client/src/controller.rs

```diff
@@ -1403,14 +1526,11 @@ where
                 Ok(export) => export,
                 Err(_) => continue,
             };
-            let from = export.from();
 
-            self.state
-                .exported_collections
-                .get_mut(&from)
-                // Internal logic error NOT due to export not existing
-                .expect("Dangling exported collection")
-                .retain(|from_export_id| *from_export_id != id);
+            let read_capability = export.read_capability.clone();
+            let storage_dependencies = export.storage_dependencies.clone();
+            // TODO: Better error handling/validation?
+            let _ = self.remove_read_capabilities(read_capability, &storage_dependencies);
```

Comment:
I would either `.unwrap()` this or thread `Result` all the way up. Ignoring it is stressful

Reply:
remove the `Result` data type. The verification logic is asserting, I think problems with this always point at a bug. wdyt?

Reply:
I saw the assertion but there is also the fetching of the collection by id that had a `?`. Up to you, I'm fine with panicking. I think we would panic anyway somewhere up the stack if the id was missing